### PR TITLE
Add output plugins for Letterboxd

### PR DESCRIPTION
**Pull request by Chemondelay**
*Had to recreate the original version of this branch, so I closed the original PR and opened a new one. Copied the description across below:*

This PR includes two new output plugins for Letterboxd: `letterboxd_add` and `letterboxd_remove`. I've followed the lead of the Trakt output plugins by including them both as subclasses within the same file, `plugins/services/letterboxd_submit.py`.

Both are configured the same way:
````
tasks:
  letterboxd_[add|remove]:
    username: *****
    password: *****
    list: [watchlist|watched]
````

The plugins uses re.search() to grab the URL slug for the film from the search results page. This means that the `imdb_id` field is required, since it's the only reliable way to isolate a single film in the results. The slug is then used to fill out various POST requests for modifying the list entries.

Only the `watchlist` and `watched` lists are supported for now. Adding or removing a film from a custom list will involve messing around with Javascript commands on the list's edit page, and since I'm basically figuring out what I'm doing as I go along, I'm stuck on this for the moment. (Incidentally, if anyone can point me in the right direction on this, I'd really appreciate it.)

I've also renamed the existing Letterboxd plugin to `letterboxd_list` to distinguish it from the new ones. The name is long enough to mess up the log formatting, not sure if this is a problem.

### CODE REVIEW

File: flexget/plugins/services/letterboxd_submit.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    def on_task_output(self, task, config):
+        requests.get(base_url)
+        token = requests.cookies['com.xk72.webparts.csrf']
+        params = {'__csrf': token}
+        auth = {
+            'username': config['username'],
+            'password': config['password']
+        }
+        headers = {'Referer': base_url}
+        r = requests.post('%s/user/login.do' % base_url, data=dict(params, **auth), headers=headers)
```
Comment (stevezau) [inline flexget/plugins/services/letterboxd_submit.py]: You should be able to use task.requests here

File: flexget/plugins/services/letterboxd_submit.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                        self.log.debug('Letterboxd response: %s' % r.text)
+                    elif r.status_code == 404:
+                        self.log.error('Can\'t access film data at: `%s/film/%s`' % (base_url, film))
+                    elif r.status_code == 401:
+                        self.log.error('Authentication error. Check your Letterboxd username and password.')
+                        self.log.debug('Letterboxd response: %s' % r.text)
+                    else:
+                        self.log.error('Unknown error accessing film data on Letterboxd: %s' % r.text)
+                else:
+                    log.warning('No imdb_id found for `%s`. '  % entry['title'] + \
```
Comment (stevezau) [inline flexget/plugins/services/letterboxd_submit.py]: change to self.log?

File: flexget/plugins/services/letterboxd_submit.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    remove = None
+    log = None
+
+
+    def parse_film(self, search):
+        url = base_url + '/search/films/%s/' % search
+        film = re.search(r'\/film\/(.*?)\/', requests.get(url).content).group(1)
+
+        return film
+
```
Comment (stevezau) [inline flexget/plugins/services/letterboxd_submit.py]: PEP8 errors.. please run through PEP8 checker..

**Comment by gazpachoking** [comment]
@Chemondelay Been a while, do you want to address the comments and merge/rebase against latest develop to resolve conflicts?

**Comment by liiight** [comment]
Please reopen if relevant

**Pull request closed without merging**